python_legacy/iceberg/api/types/type_util.py

```diff
@@ -143,7 +143,30 @@ def visit(arg, visitor): # noqa: ignore=C901
 
             return visitor.list(list_var, element_result)
         elif type_var.type_id == TypeID.MAP:
-            raise NotImplementedError()
+            # TODO: How could this be improved to avoid copy-pasting the same logic for key and value fields?
+            map_var = type_var.as_nested_type().asMapType()
+            visitor.field_ids.append(map_var.key_field.field_id)
+            visitor.field_names.append(map_var.key_field.name)
+            try:
+                key_result = visit(map_var.key_type(), visitor)
+            except NotImplementedError:
```

Comment:
Are there areas where this is still thrown?

Reply:
Except for abstract classes, this error is still thrown at a few spots in the code:
![image](https://user-images.githubusercontent.com/56447460/151443193-70e229de-6e70-49ef-bed5-c4c003861633.png)

But for things related to Map, I could only find this:
- https://github.com/apache/iceberg/blob/master/python_legacy/iceberg/api/types/type_util.py#L172